tests/ui/danger_not_accepted.rs

```diff
@@ -0,0 +1,78 @@
+#![warn(clippy::danger_not_accepted)]
+
+fn main() {
+    wee();
+    waz::woo();
+
+    #[clippy::accept_danger(may_deadlock)]
+    waz::woo();
+
+    Maz.faz();
+
+    #[clippy::accept_danger(may_deadlock)]
+    Maz.faz();
+
+    #[clippy::accept_danger(may_deadlock, may_delete_system)]
+    Maz.faz();
+
+    Maz.faz2();
+
+    #[clippy::accept_danger(may_deadlock)]
+    Maz.faz2();
+
+    #[clippy::accept_danger(may_deadlock, may_delete_system)]
+    Maz.faz2();
+
+    waz::woo2();
+
+    #[clippy::accept_danger(may_deadlock)]
+    waz::woo2();
+}
+
+fn wee() {}
+
+struct Maz;
+
+#[clippy::dangerous(may_deadlock)]
+mod waz {
+    pub fn woo() {}
+
+    #[clippy::dangerous(may_deadlock, reason = "your program may deadlock in calling this function")]
+    pub fn woo2() {}
+
+    impl super::Maz {
+        #[clippy::dangerous(
+            may_delete_system,
+            reason = "calling this has a very strong chance of just deleting your computer"
+        )]
+        pub fn faz(&self) {}
+    }
+
+    impl super::FazTrait for super::Maz {
+        fn faz2(&self) {}
+    }
+}
+
+trait FazTrait {
+    #[clippy::dangerous(may_delete_system)]
+    fn faz2(&self);
+}
+
+// Edge case attr tests
+#[rustfmt::skip]
+#[clippy::dangerous(whee, woo,)]
+#[clippy::dangerous(whee, woo, reason = "sdfhsdf",)]
```

Comment:
I think the reason should be per danger, i.e. `#[clippy::dangerous(foo = "reason")]`

Bikeshedding a bit, what about something like `#[clippy::danger::foo = "reason"]`? It's more similar to `#[must_use]` that way

Reply:
Hmm. I like the `#[clippy::dangerous(a, b, reason = "foo")]` syntax because it allows you to share the same reason among multiple dangers. For example, `foo_internal_api` and `raw_foo_access` could share the reason "low level building block; use bar instead." If you want several different reasons, you could always write it as:

```rust
#[clippy::dangerous(foo, reason = "reason 1")]
#[clippy::dangerous(bar, reason = "reason 2")]
fn baz() {}
```

Reply:
Personally I don't think it will be that common to want to use multiple dangers on the same node, particularly so where the reason can be the same for all of them

Reply:
Relatedly there was discussion about whether or not the reason should be required, I don't think we fully came to an agreement on that but initially let's make it required, it can be made optional later if we want that but not the other way around